src/wasm-interpreter.h

```diff
@@ -740,7 +740,7 @@ class ModuleInstance {
     addr += curr->offset;
     trapIfGt(curr->bytes, memorySizeBytes, "bytes > memory");
     trapIfGt(addr, memorySizeBytes - curr->bytes, "highest > memory");
-    return addr;
+    return (Address)addr;
```

Comment:
addr should be in the right range here, so there should be no risk of overflow. however, if we have a `toAddress` utility function that checks for overflow, it would still be a useful assert check here, so as in the other pr I think it makes sense to do.